py-polars/polars/expr/expr.py

```diff
@@ -195,7 +195,10 @@ def __ne__(self, other: Any) -> Self:  # type: ignore[override]
         return self._from_pyexpr(self._pyexpr.neq(self._to_expr(other)._pyexpr))
 
     def __neg__(self) -> Expr:
-        return F.lit(0) - self
+        neg_expr = F.lit(0) - self
```

Comment:
Does `keep_name` work here?

Reply:
Not reliably; if you use `keep_name` when applying unary minus to something that doesn't _have_ a name, such as `lit(1)`, you'll get a different error, like so:
```python
pl.DataFrame({"x":[3, 2, 1]}).with_columns( -pl.lit(1) )
# PanicException: expected root column to keep expression name
```
Checking the `meta.output_name` and allowing it to return `None` if inapplicable seemed the most straightforward way to handle expressions that do/don't have names (and I included a literal in the unit test to cover this ;)

Reply:
Right.. I think it is good for now. Ideally we have formal `+` and `-` operations on the rust side.